wolfcrypt/src/random.c

```diff
@@ -2175,10 +2175,16 @@ int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
 
         int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
         {
-            int i;
-
-            for (i = 0; i< sz; i++) {
-               output[i] =  esp_random( );
+            word32 rand;
+            while (sz > 0) {
+                word32 len = sizeof(rand);
+                if (sz < len)
+                    len = sz;
+                /* Get one random 32-bit word from hw RNG */  
+                rand = esp_random( );
+                XMEMCPY(output, &rand, sz);
```

Comment:
Should this copy "len" bytes instead of "sz"?  The line below increments output by "len".

Reply:
Yes, you are right. It should be "len".  Updated [the line 2231 in random.c](https://github.com/miyazakh/wolfssl/blob/da6ed416be5d157c078600fb0407f6080059956b/wolfcrypt/src/random.c#L2231) too.